src/feature/hs/hs_descriptor.c

```diff
@@ -2701,14 +2701,16 @@ hs_desc_encode_descriptor,(const hs_descriptor_t *desc,
     goto err;
   }
 
-  /* Try to decode what we just encoded. Symmetry is nice! */
-  /* XXX: I need to disable this assertation for now to make the test pass.
-   * I will enable it again when I finish writing the decoding */
-  /* ret = hs_desc_decode_descriptor(*encoded_out, */
-  /*                                 desc->subcredential, NULL); */
-  /* if (BUG(ret < 0)) { */
-  /*   goto err; */
-  /* } */
+  /* Try to decode what we just encoded. Symmetry is nice!, but it is
+   * symmetric only if the client auth is disabled. That is, the descriptor
+   * cookie will be NULL. */
+  if (!descriptor_cookie) {
```

Comment:
Can't we also test-decode even if we have a descriptor cookie?

Reply:
The reason is that `hs_desc_decode_descriptor` parameter requires the client private key not the descriptor cookie and we can't find the private key using descriptor cookie.

And I think using a private key as a parameter of `hs_desc_decode_descriptor` is more reasonable because we use a private key to decode not a descriptor cookie

Reply:
Hm, I see what you mean. IIf I'm not mistaken, I think with some medium-level refactoring we could make that smooth and also have it test-decode in the client auth case as well (so that this code path gets used).

Perhaps we can open a ticket about this.